### PR TITLE
Remove unnecessary calls to AWS API during subnet deletion

### PR DESCRIPTION
**Pull request by pydctw**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
`deleteSubnets` function calls `describeVpcSubnets` that makes multiple calls to AWS API although it only needs subnet IDs. 
Refactored the code to remove unnecessary calls, DescribeRouteTables and DescribeNatGatewaysPages, to AWS API during subnet deletion.

I also set up tests for deleteSubnets, which was missing. Plan to add more test cases as part of #3074.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:
- [ ] squashed commits
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloud/services/network/subnets.go [hunk 1]
```diff
@@ -248,13 +248,13 @@ func (s *Service) deleteSubnets() error {
...
-	existing, err := s.describeVpcSubnets()
+	existing, err := s.describeSubnets()
 	if err != nil {
 		return err
 	}
 
-	for _, sn := range existing {
-		if err := s.deleteSubnet(sn.ID); err != nil {
+	for _, sn := range existing.Subnets {
+		if err := s.deleteSubnet(*sn.SubnetId); err != nil {
```
Comment (sedefsavas) [inline pkg/cloud/services/network/subnets.go:257]:
```suggestion
		if err := s.deleteSubnet(aws.StringValue(sn.SubnetId)); err != nil {
```

Comment (pydctw) [reply pkg/cloud/services/network/subnets.go]: Thanks, didn't know about this function. Updated the PR with the suggestion.

**Comment by pydctw** [comment]
Accidentally closed during my branch cleanup :) Reopening.

@sedefsavas, could you review again?

**Comment by sedefsavas** [comment]
@pydctw content hasn't changed, right?

**Comment by pydctw** [comment]
> @pydctw content hasn't changed, right?

No. Other than addressing the comment, no changes.

**Pull request merged**